src/context.py

```diff
@@ -3,11 +3,50 @@
 from config import MAILBOX
 from commonbot.utils import combine_message, send_message
 
+class ReportModal(discord.ui.Modal):
+    def __init__(self, *, message: discord.Message):
+        super().__init__(title="Report a message")
+
+        self.message: discord.Message = message
+        self.comments_input: discord.ui.TextInput = discord.ui.TextInput(
+            label="Comments",
+            placeholder="(Optional) Additional comments or context",
+            style=discord.TextStyle.long,
+            required=False,
+            max_length=250
+        )
+        self.add_item(self.comments_input)
+
+    async def on_submit(self, interaction: discord.Interaction):
+        embed: discord.Embed = discord.Embed(
+            title=f"\N{BALLOT BOX WITH BALLOT} Report from #{self.message.channel.name}",
+            colour=discord.Colour.gold(),
+            url=self.message.jump_url)
+
+        message_str: str = combine_message(self.message)
+        comments_str: str = self.comments_input.value
+
+        len_max: int = 1000
+        [
+            embed.add_field(
+                name=field[0],
+                value=field[1] if len(field[1]) < len_max else f"{field[1][:len_max-1]}…",
+                inline=field[2])
+            for field in [
+                ("Suspect", self.message.author.mention, True),
+                ("Reported by", interaction.user.mention, True),
+                (f"Sent <t:{int(self.message.created_at.timestamp())}:R>:", message_str, False),
+                ("Comments:", comments_str, False)
+            ] if field[1]
+        ]
+
+        log_chan = interaction.guild.get_channel(MAILBOX)
+        await send_message(message="", channel=log_chan, embed=embed)
```

Comment:
Do we need to use `send_message` here? It mostly exists as a wrapper to break up messages that are over Discord's character limit, but since we're just sending an embed, we could just use the native TextChannel.send function.

Reply:
I just used it for consistency, but that's sort of undermined by having to make a PR to mangle `send_message` as a prerequisite. I'll just submit with the native method